libs/core/langchain_core/graph_stores.py

```diff
@@ -0,0 +1,477 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import (
+    Any,
+    AsyncIterable,
+    ClassVar,
+    Collection,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+)
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForRetrieverRun,
+    CallbackManagerForRetrieverRun,
+)
+from langchain_core.documents import Document
+from langchain_core.load import Serializable
+from langchain_core.pydantic_v1 import Field
+from langchain_core.runnables import run_in_executor
+from langchain_core.vectorstores import VectorStore, VectorStoreRetriever
+
+
+def _has_next(iterator: Iterator) -> bool:
+    """Checks if the iterator has more elements.
+    Warning: consumes an element from the iterator"""
+    sentinel = object()
+    return next(iterator, sentinel) is not sentinel
+
+
+class Node(Serializable):
```

Comment:
What are other types of nodes that one may want to support? Existing node is similar to our Document Schema but without the id field, but I suspect that it's useful to have a distinct concept for a Node.

Reply:
The idea is to provision for future support of multi-modal.
But it's probably too soon to introduce this abstraction without a clear view of how multi-modal would work so I think I'll remove it for now to simplify and put `text` into `Node` (maybe rename to `body` ?). We are investigating another way to support multiple formats using MIME types instead of subclasses so this can be left as a future work.